Mathlib/FieldTheory/JacobsonNoether.lean

```diff
@@ -0,0 +1,215 @@
+/-
+Copyright (c) 2024 F. Nuccio, H. Zheng, W. He, S. Wu, Y. Yuan, W. Jiao. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Filippo A. E. Nuccio, Huanyu Zheng, Sihan Wu, Wanyi He, Weichen Jiao, Yi Yuan
+-/
+import Mathlib.Algebra.CharP.LinearMaps
+import Mathlib.Algebra.CharP.Subring
+import Mathlib.Algebra.Field.Power
+import Mathlib.FieldTheory.PurelyInseparable
+
+/-!
+# The Jacobson-Noether theorem
+
+This file contains a proof of the Jacobson-Noether theorem and some auxiliary lemmas.
+Here we discuss different cases of characteristics of
+the noncommutative division algebra `D` with center `k`.
+
+## Main Results
+
+- `exists_separable_mem_of_not_central` : (Jacobson-Noether theorem) For a
+  non-commutative algebraic division algebra `D` (with base ring
+  being its center `k`), then there exist an element `x` of
+  `D \ k` that is separable over its center.
+- `exists_separable_mem_of_not_central'` : (Jacobson-Noether theorem) For a
+  non-commutative algebraic division algebra `D` (with base ring
+  being a field `L`), if the center of `D` over `L` is `L`,
+  then there exist an element `x` of `D \ L` that is separable over `L`.
+
+## Notations
+
+- `D` is a noncommutative division algebra
+- `k` is the center of `D`
+
+## Implementation Notes
+
+Mathematically, `exists_separable_mem_of_not_central` and `exists_separable_mem_of_not_central'`
+are equivalent.
+
+The difference however, is that the former takes `D` as the only variable
+and fixing `D` would forces `k`. Whereas the later takes `D` and `L` as
+separate variables constrained by certain relations.
+
+## Reference
+* <https://ysharifi.wordpress.com/2011/09/30/the-jacobson-noether-theorem/>
+-/
+
+namespace JacobsonNoether
+
+variable {D : Type*} [DivisionRing D]
+
+open LinearMap
+
+local notation3 "k" => (Subring.center D)
+private def δ : D → D →ₗ[k] D := mulLeft k - mulRight k
```

Comment:
I take my share of responsibility here, I suggested that it should be private because it seems to me to be somewhat specific to the context. I agree that it could/shuold live with `mulLeft` and `mulRight`, provided we find a right name (`\delta` is probably not ok, I would go for `bra`: any idea, @mattrobball ?)

Reply:
It looks like the adjoint action of gl(V) on itself.

Reply:
Yes, this is what motivated me to call it `bra` for bracket. But perhaps @Yu-Misaka can try to use [LieDerivation.ad](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Lie/Derivation/AdjointAction.html#LieDerivation.ad) instead?

Reply:
Do we use `𝖆𝖉` currently?

Reply:
Oh! If we have something already, we should try to plug into that as much as possible.

Reply:
Thanks! @faenuccio @mattrobball I've replaced all `δ` with `LieAlgebra.ad`. However I didn't manage to find existing lemma to replace `ad_iterate_succ` (originally private lemma `δ_iterate_succ`), so I placed it down the main theorem as a `have`.

Reply:
Great, I believe the situation of this PR is very good.

Reply:
Sorry, I just noticed that `ad_iterate_succ` can be deduced directly from `Function.iterate_succ_apply'`...